cpp/src/io/orc/reader_impl.cu

```diff
@@ -178,15 +178,28 @@ class metadata {
    * @brief Filters and reads the info of only a selection of stripes
    *
    * @param[in] stripe Index of the stripe to select
+   * @param[in] max_stripe_count Number of stripes to select for stripe-based selection
+   * @param[in] stripe_indices Indices of individual stripes [max_stripe_count]
    * @param[in] row_start Starting row of the selection
    * @param[in,out] row_count Total number of rows selected
    *
    * @return List of stripe info and total number of selected rows
    **/
-  auto select_stripes(int stripe, int max_stripe_count, int &row_start, int &row_count) {
+  auto select_stripes(size_type stripe, size_type max_stripe_count, const size_type *stripe_indices,
```

Comment:
This API could be cleaned up to use a `std::vector` as well but as @harrism mentioned doesn't need to be done in this job.

Reply:
While there are subjective arguments in favor of using a `std::vector` here, there are also objective arguments against it.